code/game/objects/items/devices/paicard.dm

```diff
@@ -77,6 +77,11 @@
 			EK.forceMove(get_turf(src))
 			installed_encryptionkeys -= EK
 		recalculateChannels()
+	else if(istype(C, /obj/item/stack/nanopaste))
+		if(!pai)
+			to_chat(user, SPAN_WARNING("You cannot repair a pAI device if there's no active pAI personality installed."))
+			return
+		pai.attackby(C, user)
```

Comment:
Is there a reason to not just reroute all attacks to the pAI mob at the end of the pai card's attackby at all?

Reply:
i guess not